src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MirrorEndpointsSpec.groovy

```diff
@@ -610,36 +590,33 @@ class MirrorEndpointsSpec extends HealthCheckSpecification {
                 .withTraffgensOnBothEnds()
                 .withAtLeastNTraffgensOnSource(2)
                 .random()
-        def flow = flowHelperV2.randomFlow(swPair).tap {
-            source.innerVlanId = 100
-            destination.innerVlanId = 200
-        }
-        flowHelperV2.addFlow(flow)
+        def flow = flowFactory.getBuilder(swPair)
+                .withSourceInnerVlan(100)
+                .withDestinationInnerVlan(200)
+                .build().create()
 
         when: "Create a mirror point"
-        def mirrorTg = swPair.src.traffGens[1]
-        def mirrorEndpoint = FlowMirrorPointPayload.builder()
-                .mirrorPointId(flowHelperV2.generateFlowId())
-                .mirrorPointDirection(mirrorDirection.toString().toLowerCase())
-                .mirrorPointSwitchId(swPair.src.dpId)
-                .sinkEndpoint(FlowEndpointV2.builder().switchId(swPair.src.dpId).portNumber(mirrorTg.switchPort)
-                        .vlanId(flowHelperV2.randomVlan([flow.source.vlanId, flow.source.innerVlanId]))
-                        .build())
-                .build()
-        flowHelperV2.createMirrorPoint(flow.flowId, mirrorEndpoint)
+        def mirrorTg = swPair.src.traffGens[1] // take TG at the same switch but not used in the flow
+        def mirrorVlanId = randomVlan([flow.source.vlanId, flow.source.innerVlanId])
+        def mirrorPointEp = flow.deepCopy().source.tap {
+            it.vlanId = mirrorVlanId
+            it.portNumber = mirrorTg.switchPort
+        }
+        flow.createMirrorPoint(mirrorPointEp, mirrorDirection)
 
         then: "Mirror point is created, flow and switches are valid"
-        switchHelper.synchronizeAndCollectFixedDiscrepancies(pathHelper.getInvolvedSwitches(flow.flowId)*.getDpId()).isEmpty()
-        northbound.validateFlow(flow.flowId).each { direction -> assert direction.asExpected }
+        def flowInvolvedSwitches = flow.retrieveAllEntityPaths().getInvolvedSwitches()
+        switchHelper.synchronizeAndCollectFixedDiscrepancies(flowInvolvedSwitches).isEmpty()
+        flow.validate().each { direction -> assert direction.asExpected }
```

Comment:
the same here, please replace it with  `validateAndCollectDiscrepancies().isEmpty()`

Reply:
Done